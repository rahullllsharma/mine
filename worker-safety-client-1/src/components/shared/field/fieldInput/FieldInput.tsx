import type { ForwardedRef } from "react";
import type { InputProps } from "../../input/Input";
import type { FieldProps } from "../Field";
import { forwardRef } from "react";
import Input from "../../input/Input";
import Field from "../Field";

export type FieldInputProps = FieldProps & InputProps;

function FieldInput(
  {
    htmlFor,
    label,
    required,
    caption,
    error,
    containerClassName,
    ...inputProps
  }: FieldInputProps,
  ref: ForwardedRef<HTMLInputElement>
): JSX.Element {
  return (
    <Field
      htmlFor={htmlFor}
      label={label}
      required={required}
      caption={caption}
      error={error}
      className={containerClassName}
    >
      <Input {...inputProps} ref={ref} error={error} />
    </Field>
  );
}

export default forwardRef(FieldInput);
