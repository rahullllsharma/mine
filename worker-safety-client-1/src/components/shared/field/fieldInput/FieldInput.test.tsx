import { render } from "@testing-library/react";
import FieldInput from "./FieldInput";

describe("FieldInput", () => {
  it('should render with the "input" component', () => {
    const { asFragment } = render(
      <FieldInput
        htmlFor="lorem"
        id="lorem"
        label="Lorem ipsum"
        placeholder="Lorem ipsum"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
