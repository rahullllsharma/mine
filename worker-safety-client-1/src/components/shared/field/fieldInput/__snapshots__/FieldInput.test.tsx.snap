// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldInput should render with the "input" component 1`] = `
<DocumentFragment>
  <div>
    <label
      class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
      for="lorem"
    >
      Lorem ipsum
    </label>
    <div
      class="relative flex w-full items-center text-base font-normal text-neutral-shade-100 rounded-md border border-solid border-neutral-shade-26 rounded-md focus-within:ring-1 focus-within:ring-brand-gray-60 bg-neutral-light-100"
    >
      <input
        aria-errormessage=""
        aria-invalid="false"
        class="flex-auto rounded-md appearance-none focus:outline-none disabled:bg-neutral-light-77 read-only:cursor-default disabled:cursor-not-allowed disabled:opacity-38 min-w-0 p-2"
        id="lorem"
        placeholder="Lorem ipsum"
        type="text"
      />
    </div>
  </div>
</DocumentFragment>
`;
